BHoM_UI/Components/Adapter/Pull.cs

```diff
@@ -82,17 +83,45 @@ public override object Run(object[] inputs)
 
         [Description("Pull objects from the external software")]
         [Input("adapter", "Adapter to the external software")]
-        [Input("request", "Filter on the objects to pull")]
+        [Input("request", "Connect a type of object to Pull objects of that type." +
+            "Connect a Request to pull only objects that satisfy a certain rule.")]
         [Input("config", "Pull config")]
         [Input("active", "Execute the pull")]
-        [Output("objects","Objects pulled")]
-        public static IEnumerable<object> Pull(BHoMAdapter adapter, IRequest request = null, Dictionary<string, object> config = null, bool active = false)
+        [Output("objects", "Objects pulled")]
+        public static IEnumerable<object> Pull(BHoMAdapter adapter, IRequest request = null, 
+            PullType pullType = PullType.AdapterDefault, ActionConfig actionConfig = null, 
+            bool active = false)
         {
+            IRequest actualRequest = null;
+
+            // ---------------------------------------------//
+            // Mandatory Adapter Action set-up              //
+            //----------------------------------------------//
+            // The following are mandatory set-ups to be ALWAYS performed 
+            // before the Adapter Action is called,
+            // whether the Action is overrided at the Toolkit level or not.
+ 
             if (request == null)
-                request = new FilterRequest();
+                actualRequest = new FilterRequest();
```

Comment:
Same here. Wouldn't this and all the below, apart from the null check on the `adapter`, be safer be in the BHoMAdapter?

Reply:
As visible, I didn't add this in for this refactoring. The reason it's there is to assure we have always have a value passed to all Adapter methods.
This cannot be assured in the Adapter because we allow the override of the Adapter Actions.

Reply:
See my comment here #167 (comment)